src/libraries/Silk.Net.Windowing/Abstract/IGLSurface.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace Silk.NET.Windowing
+{
+    public interface IGLSurface : INativeGLSurface
+    {
+        ContextFlags ContextFlags { get; set; }
+        ContextProfile ContextProfile { get; set; }
+        IGLSurface? SharedContext { get; set; }
+
+        /// <summary>
+        /// Enables OpenGL support for this surface. This will create a surface upon initialization.
```

Comment:
I don't know what specifically you are talking about here, I don't even think you would want to create a gl context upon initialization, nor gles for that matter

Reply:
I was quoting the doc line specifically. GitHub API is sometimes suboptimal
> This will create a surface upon initialization.

Reply:
Really IMO it should expose a CreateGLES context and not run at initialization.

Reply:
Initialization refers to the window actually being created on the OS. You can't create an OpenGL context without a window, and you don't have a window until `Initialize` is called.

**POSSIBLY USEFUL INCOHERENT RAMBLING:** It is possible we will want to make `ISurface` an abstract base class primarily because leaving this to the implementor is a lot of trust. i.e. it would be nice if all the implementation has to do is implement `Run(Action)` and then when that method starts we do everything that `Initialize` would do, same with `Reset` at the end, make sure the actual implementation never knows about `DoUpdate` and `DoRender` (but still expose these so users can implement their own loops) [though, I have some game loop ideas that too would render these obsolete], and leave the implementation to update the events whenever it feels like it.

Reply:
Oh and to answer the question directly, this method basically says "Hey you, surface: make an OpenGL context on the window that gets created in `Initialize`". But once again there is lots of talk of removing OpenGL and Vulkan context/surface creation in core windowing itself and directly using the mechanisms in each respective ecosystem directly, rather than entrusting this into a native library such as GLFW.